src/components/connectAdvanced.js

```diff
@@ -100,203 +90,169 @@ export default function connectAdvanced(
-          this.componentDidUpdate = this.notifyNestedSubsOnComponentDidUpdate
-          this.setState(dummyState)
+        const { state, store } = value
+        const { forwardRef, props } = this.props
+        let derivedProps = this.generatedDerivedProps(state, props, store)
+        if (connectOptions.pure) {
+          return <PureWrapper derivedProps={derivedProps} forwardRef={forwardRef} />
         }
-      }
 
-      notifyNestedSubsOnComponentDidUpdate() {
-        // `componentDidUpdate` is conditionally implemented when `onStateChange` determines it
-        // needs to notify nested subs. Once called, it unimplements itself until further state
-        // changes occur. Doing it this way vs having a permanent `componentDidUpdate` that does
-        // a boolean check every time avoids an extra method call most of the time, resulting
-        // in some perf boost.
-        this.componentDidUpdate = undefined
-        this.notifyNestedSubs()
+        return <WrappedComponent {...derivedProps} ref={forwardRef} />
       }
 
-      isSubscribed() {
-        return Boolean(this.subscription) && this.subscription.isSubscribed()
-      }
+      renderWrappedComponent(value) {
+        invariant(value,
+          `Could not find "store" in the context of ` +
+          `"${displayName}". Either wrap the root component in a <Provider>, ` +
+          `or pass a custom React context provider to <Provider> and the corresponding ` +
+          `React context consumer to ${displayName} in connect options.`
+        )
+        const { state, store } = value
+        let derivedProps = this.generatedDerivedProps(state, this.props, store)
+        if (connectOptions.pure) {
+          return <PureWrapper derivedProps={derivedProps} />
+        }
 
-      addExtraProps(props) {
-        if (!withRef && !renderCountProp && !(this.propsMode && this.subscription)) return props
-        // make a shallow copy so that fields added don't leak to the original selector.
-        // this is especially important for 'ref' since that's a reference back to the component
-        // instance. a singleton memoized selector would then be holding a reference to the
-        // instance, preventing the instance from being garbage collected, and that would be bad
-        const withExtras = { ...props }
-        if (withRef) withExtras.ref = this.setWrappedInstance
-        if (renderCountProp) withExtras[renderCountProp] = this.renderCount++
-        if (this.propsMode && this.subscription) withExtras[subscriptionKey] = this.subscription
-        return withExtras
+        return <WrappedComponent {...derivedProps} />
       }
 
       render() {
-        const selector = this.selector
-        selector.shouldComponentUpdate = false
-
-        if (selector.error) {
-          throw selector.error
-        } else {
-          return createElement(WrappedComponent, this.addExtraProps(selector.props))
+        if (this.props.unstable_observedBits) {
```

Comment:
Shouldn't be any need to treat `observedBits` specially - just render it, and it likely is undefined most of the time.

Beyond that, tbh I don't think I want to even bother with that for 6.0.  I want to get this out the door, and we can investigate `observedBits` for an update.